openms-wiki/developer-faq.md

```diff
@@ -0,0 +1,491 @@
+3. Have a look at existing GUI tests, as they use the `QT TestLib` framework and not the OpenMS macros.
+
+### (Linux) All tests fail when you execute `make test`
+
+Check the `LD_LIBRARY_PATH` environment variable:
+
+You can print the `LD_LIBRARY_PATH` with `echo $LD_LIBRARY_PATH`. If your `/lib/` folder is included, check that `libOpenMS.so` is present.
+With the `ldd` command, you can show the libraries used by an executable, e.g. `ldd /bin/ClassTest_test`.
+
+### How can I speed up the compile process of OpenMS?
+
+To speed up the compile process of OpenMS, use several threads. If you have several processors/cores, you can build OpenMS classes/tests and `TOPP` tools in several threads. On Linux, use the `make option -j: make -j8 OpenMS TOPP test_build`.
+
+On Windows, Visual Studio solution files are automatically build with the `/MP` flag, such that Visual Studio uses all available cores of the machine.
+
+## Release
+
+View [Preparation of a new OpenMS release](https://github.com/OpenMS/OpenMS/wiki/Preparation-of-a-new-OpenMS-release#release_developer) to learn more about contributing to releases.
+
+## Continuous integration
+
+### How does travis work?
+
+[Travis](https://travis-ci.org/) is an automated system for continuous integration. Each new commit and pull request is automatically run through the travis build system.
+This is controlled by a `.travis.yaml` file in the source tree.
+
+### What can we do if travis times out?
+
+Try to restart travis, it sometimes hangs and since it builds on shared infrastructure, the next build may work better. This needs to be done by a OpenMS core developer.
+
+Since we use extensive caching, the build may take much longer when many files are touched and may never complete in that case (running into the travis time limit). In that case we can rebuild the cache using the following approach:
+```bash
+ $ git cherry-pick 89c5cd7f2d9d343b3d63fc6bab18e08dcd969c05
+
+ $ git push origin develop
+
+ # Now wait for the build to complete
+
+ $ git revert 89c5cd7f2d9d343b3d63fc6bab18e08dcd969c05
+
+ $ git push origin develop
+```
+
+## Working in Integrated Development Environments (IDEs)
+
+### Why are there no `source/TEST` and `source/APPLICATIONS/TOPP|UTILS` folder?
+
+All source files added to an IDE are associated with their targets. You can find the source files for each test within its own subproject. The same is true for the `TOPP` and `UTILS` classes.
+
+### How can I easily update a lot of test files (e.g., after a small format change in result files)?
+
+Using `grep` one can simply extract the lines starting with `diff FILENAME1 FILENAME2` and replace the `diff` by `copy`.
+
+### I'm getting the error "Error C2471: cannot update program database"
+
+This is a bug in Visual Studio and there is a [bug fix] (http://code.msdn.microsoft.com/KB946040) Only apply it if you encounter the error. The bug fix might have unwanted side effects!
```

Comment:
delete space after [bug fix], it is rendering incorrectly.
<img width="1078" alt="Screenshot 2022-05-21 at 8 17 36 PM" src="https://user-images.githubusercontent.com/4143778/169656817-1b9a4e7b-f3f8-4952-b428-b2cd8df52b97.png">